src/components/Tabs.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+type Props = {
+  tabs: Tab[];
+  selectedTabId: string;
+  onTabSelected: (tab: Tab) => void;
+};
+
+class Tabs extends React.Component<Props> {
+  handleTabClick = (tab: Tab) => {
+    const { selectedTabId, onTabSelected } = this.props;
+
+    if (selectedTabId !== tab.id) {
+      onTabSelected(tab);
+    }
```

Comment:
You can do this check in the `handleTabChange` function, so move this logic there and delete this unnecessary `handleTabClick`

Reply:
![image](https://user-images.githubusercontent.com/61430197/151653445-404ced84-a5d6-4b9a-b90f-0496d28cc68f.png)

i did this because the rules forbid calling a callback if the tab was not changed